clients/admin-ui/src/features/system/useSystemsTable.tsx

```diff
@@ -0,0 +1,396 @@
+      active: true,
+    };
+    const result = await createSystemGroup(payload);
+    if (isErrorResult(result)) {
+      messageApi.error(getErrorMessage(result.error));
+    } else {
+      let successMessage = `System group '${result.data.name}' created`;
+      if (result.data.systems?.length === 1) {
+        successMessage += ` with system '${result.data.systems[0]}'`;
+      } else if (result.data.systems?.length) {
+        successMessage += ` with ${result.data.systems.length} systems`;
+      }
+      messageApi.success(successMessage);
+    }
+    setCreateModalIsOpen(false);
+  };
+
+  const handleBulkAddToGroup = useCallback(
+    async (groupKey: string) => {
+      if (!selectedRowKeys) {
+        return;
+      }
+      const currentGroup = systemGroupMap[groupKey];
+      const result = await updateSystemGroup({
+        ...currentGroup,
+        systems: [
+          ...(currentGroup.systems ?? []),
+          ...selectedRowKeys.map((key) => key.toString()),
+        ],
+      });
+      if (isErrorResult(result)) {
+        messageApi.error(getErrorMessage(result.error));
+      } else {
+        messageApi.success(
+          `${selectedRowKeys?.length} systems added to group '${groupKey}'`,
+        );
+      }
+    },
+    [systemGroupMap, updateSystemGroup, selectedRowKeys, messageApi],
+  );
+
+  const groupMenuItems = useMemo(() => {
+    return (
+      allSystemGroups?.map((group) => ({
+        key: group.fides_key,
+        label: group.name,
+        onClick: () => handleBulkAddToGroup(group.fides_key),
+      })) ?? []
+    );
+  }, [allSystemGroups, handleBulkAddToGroup]);
+
+  const columns: ColumnsType<BasicSystemResponseExtended> = useMemo(() => {
+    return [
+      {
+        title: "Name",
+        dataIndex: "name",
+        key: "name",
+        render: (name: string | null, record: BasicSystemResponseExtended) => (
+          <LinkCell
+            href={`/systems/configure/${record.fides_key}`}
+            data-testid={`system-link-${record.fides_key}`}
+          >
+            {name || record.fides_key}
+          </LinkCell>
+        ),
+        width: 300,
+        ellipsis: true,
+        fixed: "left",
+      },
+      {
+        dataIndex: "system_groups",
+        key: "system_groups",
+        render: (
+          systemGroups: string[] | undefined,
+          record: BasicSystemResponseExtended,
+        ) => (
+          <SystemGroupCell
+            // @ts-ignore - TS doesn't know we filter out undefined
+            selectedGroups={
+              systemGroups
+                ?.map((key) => systemGroupMap?.[key])
+                .filter((group) => !!group) ?? []
+            }
+            allGroups={allSystemGroups!}
+            system={{ ...record, system_groups: systemGroups ?? [] }}
+            columnState={{
+              isWrapped: true,
+              isExpanded: isGroupsExpanded,
+            }}
+          />
+        ),
+        width: 400,
```

Comment:
Still need to clean up fixed widths on columns as well. It's weird to see 400 width when the only system I have is `6sense` for example.